### DISCUSSION TITLE
Export data using SingleStore jdbc?

### DISCUSSION THREAD

**Discussion by oyeliseiev-ua** [dbeaver/dbeaver#36017] [Q&A]
Is there a way to export data using the Community Edition?
I found this PR: https://github.com/dbeaver/dbeaver/pull/20334/files, but it seems like setting supports-mysql-fetch-transform=false in the Community Edition has no effect.

Thanks.

**Comment by TB-HunBoka** [comment]
Do you mean this:
![kép](https://github.com/user-attachments/assets/6c24533a-77a5-4d97-9cf1-4bddee17e920)

Right click on the query result -> export data.
Or am I misunderstanding your question?

**Reply by oyeliseiev-ua** [reply]
Right, when you click an export, dbeaver set fetchSize as Integer.MIN_Value to enable streaming, but SingleStore doesn't allow negative values the same as Mariadb jdbc and throw an exception 'wrong fetch size'

**Reply by TB-HunBoka** [reply]
I am not familiar with SingleStore, and I could not see why is the fetchSize became a negative number.
So, you got the error right after you click on the export, you have no chance to manually override the fetch size on the export tab, right?

Can you try this command?

`@export {"type": "csv", "producer": {"fetchSize" : "1000"}, "consumer": {"outputFolder": "filepath", "outputFilePattern": "data"}, "processor": {}}

select * from your_table`

You need to run the export command, then run the query you would like to export.

**Comment by oyeliseiev-ua** [comment]
SingleStore, like MariaDB, uses MySQLDataSource.java (see the PR link I sent). The negative fetch size (Integer.MIN_VALUE) is set by default for MySQL (line 53), which works for the MySQL driver but not for MariaDB or SingleStore. So, I'm wondering why setting `?supports-mysql-fetch-transform=false` has no effect in DBeaver Community Edition (Version 24.2.3.202410201725), as referenced in the PR I sent.

Anyway, we found a workaround: set the ‘Extraction section’ → ‘Extract type’ to ‘Multiple queries’.
